pkg/route/routecomponenthelpers/routecomponentcontrollerhelpers.go

```diff
@@ -0,0 +1,133 @@
+package routecomponenthelpers
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	configv1 "github.com/openshift/api/config/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	configsetterv1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/util/retry"
+
+	"github.com/openshift/library-go/pkg/certs"
+)
+
+// RouteComponentControllerHelper aggregates necessary interfaces for Custom Certs for routes functions.
+type RouteComponentControllerHelper struct {
+	secretLister  corev1listers.SecretLister
+	ingressClient configsetterv1.IngressInterface
+}
```

Comment:
keep as two separate clients in function arguments

Reply:
Will do. I'll move the whole controller here and we'll see what we can unify.